core/blockchain.go

```diff
@@ -114,6 +115,32 @@ type BlockChain struct {
 	atxi *AtxiT
 }
 
+type ChainInsertResult struct {
+	ChainInsertEvent
+	Index int
+	Error error
+}
+
+type ReceiptChainInsertResult struct {
+	ReceiptChainInsertEvent
+	Index int
+	Error error
+}
+
+type HeaderChainInsertResult struct {
+	HeaderChainInsertEvent
+	Index int
+	Error error
+}
+
+func (bc *BlockChain) GetHeaderByHash(h common.Hash) *types.Header {
```

Comment:
These are going to be useful for when we do batch writes. They satisfy the interfaces that our new downloader likes.